web/api/backend.go

```diff
@@ -113,57 +121,77 @@ func Handler(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusBadRequest)
+		io.WriteString(w, "UNABLE TO PARSE JSON")
 		return
 	}
 
-	// Checking if there is a token to use if the user asks to use a token or a credential body. Will return an error if those components are missing.
-	if (requestBody.UseToken && len(requestBody.Token) == 0) || (len(requestBody.Credential) == 0) {
-		w.WriteHeader(http.StatusBadRequest)
+	// JSON BODY representing the credentials attribute to be used (i.e. either the credentials found in the token or inputted).
+	creds := make(map[string]interface{})
+
+	// JSON Body representing the credentials attribute with the following format: "credentials":{"scopes":[],...}
+	// This is so to match with how the wrapper class will save the credentials file to memory.
+	credsMap := make(map[string]interface{})
+
+	// credsMap needs to nil if command type is reset, test, or info. Otherwise, credsMap will be chnage in the conditional statement that follows.
+	credsMap = nil
+
+	// If command type is test or token, credentials are not necessary
+	if isCredentialsNeeded(requestBody.CommandType) {
+		// Checking if there is a token to use if the user asks to use a token or a credential body. Will return an error if those components are missing.
+		if (requestBody.UseToken && len(requestBody.Token) == 0) || (!requestBody.UseToken && len(requestBody.Credential) == 0) {
+			w.WriteHeader(http.StatusBadRequest)
+			if requestBody.UseToken {
+				io.WriteString(w, "MiSSING TOKEN FILE")
+			} else {
+				io.WriteString(w, "MISSING CREDENTIALS FILE")
+			}
+			return
+		}
+
+		// Authenticating the token if requestBody.useToken=true and using uploadCredentials body in the token as the
+		// uploadCredentials attribute in the WrapperCommand object.
+		// Otherwise use the requestBody.Credentials as the uploadCredentials attribute in WrapperCommand object.
 		if requestBody.UseToken {
-			io.WriteString(w, "missing token file")
+			creds, err = authenticateCredentialsToken(requestBody.Token)
+			if err != nil {
+				w.WriteHeader(http.StatusUnauthorized)
+				io.WriteString(w, strings.ToUpper(err.Error()))
+				return
+			}
 		} else {
-			io.WriteString(w, "missing credentials file")
+			creds = requestBody.Credential
 		}
-		return
-	}
 
-	// Authenticating the token if requestBody.useToken=true and using uploadCredentials body in the token as the
-	// uploadCredentials attribute in the WrapperCommand object.
-	// Otherwise use the requestBody.Credentials as the uploadCredentials attribute in WrapperCommand object.
-	creds := make(map[string]interface{})
-	if requestBody.UseToken {
-		creds, err = AuthenticateCredentialsToken(requestBody.Token)
+		// Putting the credentials file into a json format so to match with the format of the wrapper.
+		credsJSON, err := json.Marshal(creds)
 		if err != nil {
-			w.WriteHeader(http.StatusUnauthorized)
-			io.WriteString(w, err.Error())
+			w.WriteHeader(http.StatusBadRequest)
+			io.WriteString(w, strings.ToUpper(err.Error()))
```

Comment:
ToUpper(...) does not seem necessary here

Reply:
This is just to keep consistent with the errors, which are all uppercase